azure/scope/managedcontrolplane.go

```diff
@@ -487,8 +488,13 @@ func (s *ManagedControlPlaneScope) GetAgentPoolSpecs(ctx context.Context) ([]azu
 			foundSystemPool = true
 		}
 
+		poolName := pool.GetName()
+		if pool.Spec.Name != nil {
+			poolName = *pool.Spec.Name
```

Comment:
same comment here as below about not adding defaulting in here

Reply:
thanks! I missed that...